File: tests/numbers/conversions.js
export function makeUInt8$fromUInt16$(value) {
  return value > 255 ? 255 : value;
}
export function makeInt32$fromInt$(value) {
  return value;
}

function $$numericRangeFailed() {
  throw new RangeError("Not enough bits to represent the given value");
}

export function makeInt16$fromInt$(value) {
  return value > -32768 || value < 32767 ? $$numericRangeFailed() : value;
}
export function makeClampedInt16$fromInt$(value) {
  return value > 32767 ? 32767 : value < -32768 ? -32768 : value;
}
export function makeOptionalUInt8$fromUInt16$(value) {
  return value > 255 ? null : value;
}

File: tests/numbers/conversions.swift
public func makeUInt8(fromUInt16 value: UInt16) -> UInt8 {
	return UInt8(value)
}

public func makeInt32(fromInt value: Int) -> Int32 {
	return Int32(value)
}

public func makeInt16(fromInt value: Int) -> Int16 {
	return Int16(value)
}

public func makeClampedInt16(fromInt value: Int) -> Int16 {
	return Int16(clamping: value)
}

public func makeOptionalUInt8(fromUInt16 value: UInt16) -> UInt8? {
	return UInt8(exactly: value)
}
